src/transport/SecureSessionMgr.cpp

```diff
@@ -431,7 +436,41 @@ void SecureSessionMgr::SecureMessageDispatch(const PacketHeader & packetHeader,
 
     if (packetHeader.GetDestinationNodeId().HasValue())
     {
-        admin->SetNodeId(packetHeader.GetDestinationNodeId().Value());
+        nodeId = packetHeader.GetDestinationNodeId().Value();
+        if (nodeId != kUndefinedNodeId && admin->GetNodeId() != nodeId)
+        {
+            admin->SetNodeId(nodeId);
+            ChipLogProgress(Inet, "Setting nodeID %" PRIX64 " on admin.", admin->GetNodeId());
+            modifiedAdmin = 1;
+        }
+        
+    }
+
+    if (packetHeader.GetSourceNodeId().HasValue())
+    {
+        fabricId = packetHeader.GetSourceNodeId().Value();
+        if (fabricId != kUndefinedFabricId && admin->GetFabricId() != fabricId)
+        {
+            admin->SetFabricId(packetHeader.GetSourceNodeId().Value());
+            ChipLogProgress(Inet, "Setting fabricID %" PRIX64 " on admin.", admin->GetFabricId());
```

Comment:
I don't understand what this code is trying to do and why.

Reply:
Left comments/documentation